File: java/src/formula/interval/ListInterval.java
package formula.interval;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Iterator;

import scheme.Interval;


public class ListInterval implements Iterable<Interval>{
	
	protected ArrayList<Interval> listInterval;
	
	public Iterator<Interval> iterator() {
		return listInterval.iterator();
	}
	
	public ListInterval() {
		listInterval = new ArrayList<Interval>();
	}
	
	public ListInterval(ArrayList<Interval> listInterval) {
		this.listInterval = listInterval;
	}
	
	public void add(Interval intv) {
		listInterval.add(intv);
	}
	
	public int size() {
		return listInterval.size();
	}
	
	public void remove (int index) {
		listInterval.remove(index);
	}

	public Interval get(int index) {
		return listInterval.get(index);
	}
	
	public void sortBy(ListIntervalFormula lif) {

		for (Interval intv : listInterval) {
			intv.setComparatorsIIF(lif);			
		}
		Collections.sort(listInterval);
	}
	
	public ListInterval getSortedInterval(int nbIntv, ListIntervalFormula lif ) {		
		ListInterval il = new ListInterval();
		
		this.sortBy(lif);

		for (int i = 0; i < nbIntv; i++) {	
			if (i < listInterval.size())
				il.add(listInterval.get(i));
		}
		return il;
	}
	
	public void removeIfStatNotEqual(IIntervalFormula intvF, double value) {
		ArrayList<Interval> newList = new ArrayList<Interval>();
		
		for (Interval intV : this.listInterval) {	
			if (intV.getStat(intvF) == value) {
				newList.add(intV);
			}
		}
		this.listInterval = newList;
	}
	
	public void removeIfStatEqual(IIntervalFormula intvF, double value) {
		ArrayList<Interval> newList = new ArrayList<Interval>();
		
		for (Interval intV : this.listInterval) {	
			if (intV.getStat(intvF) != value) {
				newList.add(intV);
			}
		}
		this.listInterval = newList;
	}
	

}

File: php/Scheme/Formula/Interval/ListInterval.php
<?php

namespace Scheme\Formula\Interval;

use Scheme\Interval;

class ListInterval {
	
  protected $listInterval;
		
  public function __construct(array $listInterval = []) {
    if (count($listInterval)) {
      $this->listInterval = $listInterval;
    } else {
      $this->listInterval = [];
    }
  }
	
  public function add(Interval $intv) {
    $this->listInterval[] = $intv;
  }
	
  public function size() {
    return count($this->listInterval);
  }
	
  public function remove ($idx) {
    unset($this->listInterval[$idx]);
  }

  public function get($index) {
    return $this->listInterval[$index];
  }
	
  public function sortBy(ListIntervalFormula $lif) {

    foreach ($this->listInterval as $key => $interval) {
      $interval->setComparatorsIIF($lif);			
    }
    usort($this->listInterval, ["Scheme\Interval", "compareTo"]);
  }
	
  public function getSortedInterval($nbIntv, ListIntervalFormula $lif) {		
    $il = new ListInterval();
		
    $this->sortBy($lif);

    for ($i = 0; $i < $nbIntv; $i++) {	
      if ($i < count($listInterval)) {
	$il->add($this->listInterval->get($i));
      }
    }
    return $il;
  }
	

}
